### ISSUE TITLE
Cache header in refreshExpiredIdToken breaks Cloudflare Workers

### ISSUE THREAD

**Issue by AlecKriebel** [awinogrodzki/next-firebase-auth-edge#74]
Unfortunately, Cloudflare Workers [do not support setting the `cache` header on requests](https://github.com/cloudflare/workerd/issues/698). Because of this, when attempting to refresh tokens within my Next.js application deployed using [next-on-pages](https://github.com/cloudflare/next-on-pages), I get the following error: 
```
Error: The 'cache' field on 'RequestInitializerDict' is not implemented.
    at w (__next-on-pages-dist__/chunks/18809.js:2:301)
    at f (__next-on-pages-dist__/functions/middleware.func.js:1:38647)
    at le (__next-on-pages-dist__/functions/middleware.func.js:1:39831)
    at __next-on-pages-dist__/functions/middleware.func.js:1:74663
    at M (__next-on-pages-dist__/functions/middleware.func.js:1:39596)
    at async se (__next-on-pages-dist__/functions/middleware.func.js:8:4387)
    at async Module.<anonymous> (__next-on-pages-dist__/functions/middleware.func.js:22:76050)
    at async j (bundledWorker-0.5543014657482794.mjs:311:13)
    at async w.runRouteMiddleware (bundledWorker-0.5543014657482794.mjs:386:13)
    at async w.checkRoute (bundledWorker-0.5543014657482794.mjs:425:69)
```
I believe this is caused by setting a cache header in this line: https://github.com/awinogrodzki/next-firebase-auth-edge/blob/main/src/auth/index.ts#L98

I've not yet confirmed this is the root cause of the issue, but am attempting to do so by removing this line in a forked version of the package here: https://github.com/AlecKriebel/next-firebase-auth-edge/commit/f3874e280266374fdc562dec9ac24c22c760fb3d and will report back with results. 

I recognize this is a really unfortunate issue with Cloudflare workers, but having support for this, possibly through a new boolean config parameter, would be amazing!

**Comment by awinogrodzki** [comment]
Hey @AlecKriebel!

Thank you for reporting and testing the issue. If it turns out to be `cache` header problem, let me know and we'll figure something out.

**Comment by AlecKriebel** [comment]
Confirmed. 

I've removed the cache header from this package here: https://github.com/AlecKriebel/next-firebase-auth-edge/commit/f3874e280266374fdc562dec9ac24c22c760fb3d and pulled that into my application. 

Before (with an auth token that has expired): 
```bash
curl --location 'https://d470c050.starter-c0a.pages.dev/api/orgs' \
--header 'Cookie: AuthToken=<REDACTED>; AuthToken.sig=Xu3E0UleEqepu9AgEzPk7LTUxhY'

Internal Server Error          
```

With the updated `next-firebase-auth-edge` without the cache header (with an auth token that has expired): 
```bash
url --location 'https://627ffbd3.starter-c0a.pages.dev/api/orgs' \
--header 'Cookie: AuthToken=<REDACTED>; AuthToken.sig=Xu3E0UleEqepu9AgEzPk7LTUxhY'

[{"id":"0o3gdDGhNrNCmiKdJoGfx","name":"yeet2","plan":"free","stripeCustomerId":"cus_OMQMul0lx3PEv8","createdAt":"2023-07-16T06:39:04.000Z","updatedAt":"2023-07-16T06:39:04.000Z"},{"id":"mIc7ES0QUXy3_9kyCQLmF","name":"yeet","plan":"paid","stripeCustomerId":"cus_OMMMISs9J7Xrle","createdAt":"2023-07-17T05:34:57.000Z","updatedAt":"2023-07-17T05:34:57.000Z"},{"id":"HjfJ6oA37bZE4BsRbstex","name":"test2","plan":"free","stripeCustomerId":null,"createdAt":"2023-07-19T15:40:00.000Z","updatedAt":"2023-07-19T15:40:00.000Z"},{"id":"RiXfwHYjHwveKgmb_qnQf","name":"test","plan":"free","stripeCustomerId":null,"createdAt":"2023-07-17T05:04:52.000Z","updatedAt":"2023-07-17T05:04:52.000Z"},{"id":"3z9lxG0C72tQbN2pe6faX","name":"can-you-see","plan":"free","stripeCustomerId":null,"createdAt":"2023-07-19T05:26:55.000Z","updatedAt":"2023-07-19T05:26:55.000Z"},{"id":"iPiKtEdSK0hDEorvExbX-","name":"aleckriebel-gmail","plan":"free","stripeCustomerId":null,"createdAt":"2023-07-23T12:59:24.000Z","updatedAt":"2023-07-23T12:59:24.000Z"},{"id":"IBKChEazBEXlQyvOPZK0G","name":"aleckriebel","plan":"free","stripeCustomerId":null,"createdAt":"2023-07-23T12:57:15.000Z","updatedAt":"2023-07-23T12:57:15.000Z"}]
```

You can reproduce this by:
- Getting an auth cookie by signing in at https://starter-c0a.pages.dev/
- Waiting an hour for the token to expire
- Using those cookies (both `AuthToken` and `AuthToken.sig`) to do a GET against `https://d470c050.starter-c0a.pages.dev/api/orgs` => should 500 internal server error
- Using those cookies (both `AuthToken` and `AuthToken.sig`) to do a GET against `https://627ffbd3.starter-c0a.pages.dev/api/orgs` (this version has the cache header removed) => should 200 success

@awinogrodzki I'd be more than happy to write a PR for whatever the fix needs to be, but wanted to see if you had any suggestions before I get to writing. Would adding a new boolean property onto `CreateAuthMiddlewareOptions` to control if this header gets set or not make sense?

**Comment by Enalmada** [comment]
I can confirm the same error with cache header on cloudflare next-on-pages:
```
[mf:inf] GET /app 200 OK (167ms)
ApolloError: The 'cache' field on 'RequestInitializerDict' is not implemented.
    at new s (__next-on-pages-dist__/functions/app.func.js:170:157)
    at __next-on-pages-dist__/functions/app.func.js:170:38037
    at R (__next-on-pages-dist__/functions/app.func.js:176:34140)
    at __next-on-pages-dist__/functions/app.func.js:176:34066
    at new Promise (<anonymous>)
    at Object.then (__next-on-pages-dist__/functions/app.func.js:176:34033)
    at Object.error (__next-on-pages-dist__/functions/app.func.js:176:34150)
    at A (__next-on-pages-dist__/functions/app.func.js:355:1919)
    at O (__next-on-pages-dist__/functions/app.func.js:355:2282)
    at U.error (__next-on-pages-dist__/functions/app.func.js:355:2844) {
  name: ApolloError,
  graphQLErrors: Array(0),
  protocolErrors: Array(0),
  clientErrors: Array(0),
  networkError: Error: The 'cache' field on 'RequestInitializerDict' is not implemented.
    at globalThis.fetch (m…
  ...
}
```

I worked around it for now using `pnpm patch next-firebase-auth-edge` to modify the source.

**Comment by awinogrodzki** [comment]
Hey @Enalmada @AlecKriebel,

I initially set `cache: no-store` header to disable Next.js cache when refreshing tokens, but turns out it's not needed. 

According to https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating cache is disabled automatically just by setting `Content-Type` header.

Fix was released in `next-firebase-auth-edge@0.8.1`

**Comment by AlecKriebel** [comment]
Thanks so much @awinogrodzki!
